src/Files.App/MainWindow.xaml.cs

```diff
@@ -56,9 +56,26 @@ private void EnsureEarlyWindow()
 
 			// Workaround for full screen window messing up the taskbar
 			// https://github.com/microsoft/microsoft-ui-xaml/issues/8431
-			InteropHelpers.SetPropW(WindowHandle, "NonRudeHWND", new IntPtr(1));
+			// This property should only be set if the "Automatically hide the taskbar in desktop mode" setting is enabled.
+			// Setting this property when the setting is disabled will result in the taskbar overlapping the application
+			if (IsAutoHideTaskbarEnabled()) 
+				InteropHelpers.SetPropW(WindowHandle, "NonRudeHWND", new IntPtr(1));
 		}
 
+		private static bool IsAutoHideTaskbarEnabled()
```

Comment:
This should probably go in a helper
Fyi @0x5bfa

Reply:
Yes. However, we still don't have a proper class for something like this, so `AppLifecycleHelper` class.

Reply:
@Qin-shihuang Please move `IsAutoHideTaskbarEnabled` class to that helper class to make `MainWindow` class slim.